Fix bug with crawler ignoring certain collection files

qchaupds opened the pull request:
1. Add extra bundle file to observe crawler ignoring it :src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml \
2. Add extra collection to observe crawler ignoring it :src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml \
3. Add extra collection to observe crawler processing it :src/test/resources/github69/invalid/data_spectra/spectra_doc_collection_inventory_2.xml \
4. Add logic to ignore certain file and keep certain files based on logical_identifier values :src/main/java/gov/nasa/pds/tools/validate/BundleManager.java
5. Remove unused functions findAllBundleFiles and findAllCollectionFiles() :src/main/java/gov/nasa/pds/tools/validate/BundleManager.java

Refs:

https://github.com/nasa-pds/validate/issues/372 Issues with logic for reading latest version of collections, and file read logging lost with v2.* of validate

Closes:

https://github.com/nasa-pds/validate/issues/372 Issues with logic for reading latest version of collections, and file read logging lost with v2.* of validate

Tested in DEV:

Since issue #372 is a follow on issue #69, there we will be a re-running github69 invalid test to see the the 2nd "spectra_doc" collection file is now NOT ignored.
Note that github69 resources directory now have 3 new files added:

src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml
src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml
src/test/resources/github69/invalid/data_spectra/spectra_doc_collection_inventory_2.xml

The log file will show that these files will be SKIP and the reason.

src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml
src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml

% validate -R pds4.bundle --skip-content-validation -r report_github69_bundle_invalid.json   -s json -t src/test/resources/github69/invalid/bundle_kaguya_derived.xml >& t10

% grep "SKIP:" t10

[main] INFO gov.nasa.pds.tools.validate.BundleManager - SKIP: file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml due to not being selected as the bundle target
[main] INFO gov.nasa.pds.tools.validate.BundleManager - SKIP: file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml due to logical_identifier the same as selected collection but different version

Commit:
Fix bug with crawler ignoring certain collection files

1. Add extra bundle file to observe crawler ignoring it :src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml \
2. Add extra collection to observe crawler ignoring it :src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml \
3. Add extra collection to observe crawler processing it :src/test/resources/github69/invalid/data_spectra/spectra_doc_collection_inventory_2.xml \
4. Add logic to ignore certain file and keep certain files based on logical_identifier values :src/main/java/gov/nasa/pds/tools/validate/BundleManager.java
5. Remove unused functions findAllBundleFiles and findAllCollectionFiles() :src/main/java/gov/nasa/pds/tools/validate/BundleManager.java

Refs:

https://github.com/nasa-pds/validate/issues/372 Issues with logic for reading latest version of collections, and file read logging lost with v2.* of validate

Commit:
Add check for null-ness of otherBundleFiles before attempting to get to the size.

jordanpadams commented on src/main/java/gov/nasa/pds/tools/validate/BundleManager.java:
```diff
@@ -332,6 +291,66 @@ public static Target getLatestBundle() {
         return(children);
     }
 
+    private static ArrayList<Target> _getCollectionFilesWithSameLogicalIdentifier(List<Target> latestCollectionList, List<Target> otherCollectionList) {
```
@qchaupds sorry to nitpick so much here, but the `_get*` method naming is introducing a new style that we have not been using throughout the code. if there is a good reason for it, I am all in. but i would prefer to we stick to what we have been doing until we are able to spend some significant time refactoring the code.

qchaupds replied:
The underscore has been removed:

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 150 % grep "getCollectionFilesWithSameLogicalIdentifier("  ./src/main/java/gov/nasa/pds/tools/validate/BundleManager.java
    private static ArrayList<Target> getCollectionFilesWithSameLogicalIdentifier(List<Target> latestCollectionList, List<Target> otherCollectionList) {
            ignoreCollectionList =

jordanpadams reviewed (changes requested):
@qchaupds I am not seeing the changes you are mentioning in the PR log? this is here is the output I am getting from the test case I was running:

```
$ validate-2.1.0-SNAPSHOT/bin/validate --skip-content-validation -R pds4.bundle -t src/test/resources/github69/invalid/bundle_kaguya_derived.xml --skip-context-reference-check

PDS Validate Tool Report

Configuration:
   Version                       2.1.0-SNAPSHOT
   Date                          2021-08-12T17:11:51Z

Parameters:
   Targets                       [file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/bundle_kaguya_derived.xml]
   Rule Type                     pds4.bundle
   Severity Level                WARNING
   Recurse Directories           true
   File Filters Used             [*.xml, *.XML]
   Data Content Validation       off
   Product Level Validation      on
   Allow Unlabeled Files         false
   Max Errors                    100000
   Registered Contexts File      /Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/validate-2.1.0-SNAPSHOT/resources/registered_context_products.json



Product Level Validation Results

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/bundle_kaguya_derived.xml
        1 product validation(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/kgrs_calibrated_spectra_per4.xml
        2 product validation(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/kgrs_calibrated_spectra_per1.xml
        3 product validation(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory.xml
        4 product validation(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/kgrs_calibrated_spectra_per3.xml
        5 product validation(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/kgrs_calibrated_spectra_per2.xml
        6 product validation(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_doc_collection_inventory_2.xml
        7 product validation(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_doc_collection_inventory.xml
        8 product validation(s) completed


PDS4 Bundle Level Validation Results

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory.xml
        1 integrity check(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_doc_collection_inventory_2.xml
      WARNING  [warning.integrity.member_not_found]   The member 'urn:nasa:pds:insight_documents:document_rise:trk-2-34-revn-l5_tnf::1.0' could not be found in any product within the given target.
      WARNING  [warning.integrity.unreferenced_member]   Identifier 'urn:nasa:pds:insight_documents:document_rise::2.0' is not a member of any collection within the given target
        2 integrity check(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_doc_collection_inventory.xml
      WARNING  [warning.integrity.member_not_found]   The member 'urn:nasa:pds:insight_documents:document_rise:trk-2-34-revn-l5_tnf::1.0' could not be found in any product within the given target.
      WARNING  [warning.integrity.unreferenced_member]   Identifier 'urn:nasa:pds:insight_documents:document_rise::1.0' is not a member of any collection within the given target
        3 integrity check(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/bundle_kaguya_derived.xml
        4 integrity check(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/kgrs_calibrated_spectra_per1.xml
        5 integrity check(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/kgrs_calibrated_spectra_per2.xml
        6 integrity check(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/kgrs_calibrated_spectra_per3.xml
        7 integrity check(s) completed

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/kgrs_calibrated_spectra_per4.xml
        8 integrity check(s) completed

Summary:

  0 error(s)
  4 warning(s)

  Product Validation Summary:
    8          product(s) passed
    0          product(s) failed
    0          product(s) skipped

  Referential Integrity Check Summary:
    8          check(s) passed
    0          check(s) failed
    0          check(s) skipped

  Message Types:
    2            warning.integrity.member_not_found
    2            warning.integrity.unreferenced_member

End of Report
Completed execution in 4825 ms


```

Commit:
Remove leading underscore in function name

jordanpadams commented on src/main/java/gov/nasa/pds/tools/validate/BundleManager.java:
```diff
@@ -504,10 +532,24 @@ public static void makeException(URL url, String location) {
+            // Only print the size if we know that otherBundleFiles is not null.
+            LOG.debug("makeException:url,otherBundleFiles.size() {},{}",url,otherBundleFiles.size());
+            for (Target target : otherBundleFiles) {
+                LOG.info("SKIP: {} due to not being selected as the bundle target",target.getUrl());
```
@qchaupds i see now that this is not appearing because this is being logged as an info message. I was hoping we would leverage the [SKIP status](https://github.com/NASA-PDS/validate/blob/main/src/main/java/gov/nasa/pds/validate/status/Status.java#L42) and log that in our logs as appropriate. looks like we use that here: https://github.com/NASA-PDS/validate/blob/main/src/main/java/gov/nasa/pds/validate/ValidateLauncher.java#L1344 . So something like that?

qchaupds replied:
@jordanpadams Please grab the lastest commit that has changes to 3 files to implement what you'd suggested.

It is odd that you weren't able to see any messages.  I am able to see the "SKIP:" both from the INFO logging and the System.println statements.  Perhaps my environment is different than yours.  Regardless, you should be able to see the  "SKIP:" messages now.

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 130 % validate -R pds4.bundle -v 0 --skip-context-reference-check --skip-content-validation -r report_github372_bundle_invalid.json -s json -t src/test/resources/github69/invalid/bundle_kaguya_derived.xml > & t10

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 131 % grep SKIP: t10                                                                                                                                  [main] INFO gov.nasa.pds.tools.validate.BundleManager - SKIP: file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml due to not being selected as the bundle target
SKIP: file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml due to not being selected as the bundle target
[main] INFO gov.nasa.pds.tools.validate.BundleManager - SKIP: file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml due to logical_identifier the same as selected collection but different version
SKIP: file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml due to logical_identifier the same as selected collection but different version

jordanpadams replied:
@qchaupds sorry this is not clear. Instead of adding these custom "SKIP" print statements or any of the other methods that are being custom built here, I would like us to use the the existing SKIP Status enumeration to actually tag these files as "SKIPPED" in the output reporting. https://github.com/NASA-PDS/validate/blob/main/src/main/java/gov/nasa/pds/validate/status/Status.java#L42 

By doing so, these will be reported out as part of our nominal reporting mechanisms (e.g. no change in verbosity needed, no debug flags needed, etc.). Here is a notional example of what I am thinking will be output when running validate with our nominal reporting. Note the output, and the count changes in the reporting:

**With no verbosity set (e.g. default output)**
```
$ validate-2.1.0-SNAPSHOT/bin/validate --skip-content-validation -R pds4.bundle -t src/test/resources/github69/invalid/bundle_kaguya_derived.xml
```
```
...
  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory.xml
        4 product validation(s) completed

  SKIP: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml
        5 product validation(s) completed

  SKIP: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml
        6 product validation(s) completed
...
  Product Validation Summary:
    8          product(s) passed
    0          product(s) failed
    2          product(s) skipped

  Referential Integrity Check Summary:
    8          check(s) passed
    0          check(s) failed
    2          check(s) skipped
```

**With verbosity set to INFO to see why the files were skipped**
```
$ validate-2.1.0-SNAPSHOT/bin/validate --skip-content-validation -R pds4.bundle -v 1 -t src/test/resources/github69/invalid/bundle_kaguya_derived.xml
```
```
...
  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory.xml
        ...
        4 product validation(s) completed

  SKIP: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml
        INFO  [info.integrity.unreferenced_file]  Skipping file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml
        due to not being selected as the bundle target
        5 product validation(s) completed

  SKIP: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml
        INFO  [info.integrity.unreferenced_file]  Skipping file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml
        due to logical_identifier the same as selected collection but different version
        6 product validation(s) completed
...
  Product Validation Summary:
    8          product(s) passed
    0          product(s) failed
    2          product(s) skipped

  Referential Integrity Check Summary:
    8          check(s) passed
    0          check(s) failed
    2          check(s) skipped
```

qchaupds replied:
@jordanpadams I am not able to see the difference between the "validate" commands for the two examples.
Are you missing the "-v 1" or something like that for the 2nd example?

jordanpadams replied:
Yep. Sorry. Just updated below.

Commit:
Responses to comments from pull request: Add capture of -v value and print "SKIP:" statements if debug is set.

1. Add severity variable to house keeping :src/main/java/gov/nasa/pds/tools/util/FlagsUtil.java
2. Add print statement of "SKIP:"  :src/main/java/gov/nasa/pds/tools/validate/BundleManager.java
3. Add call to FlagsUtil.setSeverity() :src/main/java/gov/nasa/pds/validate/ValidateLauncher.java

Refs:

https://github.com/nasa-pds/validate/issues/372 Issues with logic for reading latest version of collections, and file read logging lost with v2.* of validate

Commit:
Second responses to comments from pull request: Add reporting of files skipped in report instead of standard out and correct log level from error to debug

1. Add setReport() so a problem can be added to report :src/main/java/gov/nasa/pds/tools/label/LocationValidator.java \
2. Correct logging level from error to debug :src/main/java/gov/nasa/pds/tools/util/ReferentialIntegrityUtil.java \
3. Add reporting of files skipped in report instead of standard out :src/main/java/gov/nasa/pds/tools/validate/BundleManager.java \
4. Add new type UNREFERENCED_FILE for use when files are skipped :src/main/java/gov/nasa/pds/tools/validate/ProblemType.java \
5. Add call to setReport() to set report in LocationValidator object :src/main/java/gov/nasa/pds/validate/ValidateLauncher.java \
6. Add logging capability :src/main/java/gov/nasa/pds/validate/report/Report.java

Refs:

https://github.com/nasa-pds/validate/issues/372 Issues with logic for reading latest version of collections, and file read logging lost with v2.* of validate

jordanpadams commented on src/main/java/gov/nasa/pds/tools/validate/BundleManager.java:
```diff
@@ -504,10 +551,56 @@ public static void makeException(URL url, String location) {
+                            //   FATAL(1,"FATAL_ERROR");
+                            // In our case, we want ExceptionType to be WARNING so it gets printed in the report for default verbose level of 3 (WARNING)
+                            ValidationProblem p1 = null;
+                            if ((FlagsUtil.getSeverity() != null) &&
```
@qchaupds i think we may be over-complicating this a bit? I do not think we need any custom logic for handling things based upon the severity set in the parent or anything like that. apologies if that was not clear in my previous example. I think this should just be:

```
                                p1 = new ValidationProblem(new ProblemDefinition(ExceptionType.INFO,
                                                               ProblemType.UNREFERENCED_FILE, "Skipping " + target.getUrl() + " due to not being selected as the bundle target"),
                                                           target.getUrl());
```

no if/then logic necessary based upon severity. all of that is handled when this is reported out.

jordanpadams commented on src/main/java/gov/nasa/pds/tools/validate/BundleManager.java:
```diff
@@ -533,6 +626,38 @@ public static void makeException(URL url, String location) {
+                            // In our case, we want ExceptionType to be WARNING so it gets printed in the report for default verbose level of 3 (WARNING)
+                            ValidationProblem p1 = null;
+                            if ((FlagsUtil.getSeverity() != null) &&
+                                (FlagsUtil.getSeverity().isInfoApplicable() || FlagsUtil.getSeverity().isDebugApplicable())) {
```
@qchaupds similar to previous comment, I may be wrong, but I don't think we need any of this custom logic to handle this reporting. Let's just use the existing reporting logic based upon the set `ExceptionType`. In this case, the `ExceptionType` should be changed to `ExceptionType.INFO`.

qchaupds replied:
@jordanpadams Per your comment, the code has been simplified in the parent and left the logic what to print in the Report.java class.    Please have another look at the last code commit.

Re-run the 3 tests resulted in the following "SKIP" status:
```
validate -R pds4.bundle --skip-context-reference-check --skip-content-validation -r report_github372_bundle_invalid_without_v.json   -s json -t src/test/resources/github69/invalid/bundle_kaguya_derived.xml >& t10

validate -R pds4.bundle -v 1 --skip-context-reference-check --skip-content-validation -r report_github372_bundle_invalid_with_v_1.json   -s json -t src/test/resources/github69/invalid/bundle_kaguya_derived.xml >& t11

validate -R pds4.bundle -v 0 --skip-context-reference-check --skip-content-validation -r report_github372_bundle_invalid_with_v_0.json   -s json -t src/test/resources/github69/invalid/bundle_kaguya_derived.xml >& t12


grep -A 6 SKIP report_github372_bundle_invalid_with*.json
report_github372_bundle_invalid_without_v.json:      "status": "SKIP",
report_github372_bundle_invalid_without_v.json-      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml",
report_github372_bundle_invalid_without_v.json-      "messages": [
report_github372_bundle_invalid_without_v.json-        {
report_github372_bundle_invalid_without_v.json-          "severity": "WARNING",
report_github372_bundle_invalid_without_v.json-          "type": "info.integrity.unreferenced_file",
report_github372_bundle_invalid_without_v.json-          "message": "Skipping file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml due to not being selected as the bundle target"
--
report_github372_bundle_invalid_without_v.json:      "status": "SKIP",
report_github372_bundle_invalid_without_v.json-      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml",
report_github372_bundle_invalid_without_v.json-      "messages": [
report_github372_bundle_invalid_without_v.json-        {
report_github372_bundle_invalid_without_v.json-          "severity": "WARNING",
report_github372_bundle_invalid_without_v.json-          "type": "info.integrity.unreferenced_file",
report_github372_bundle_invalid_without_v.json-          "message": "Skipping file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml due to logical_identifier the same as selected collection but different version"
--
report_github372_bundle_invalid_with_v_0.json:      "status": "SKIP",
report_github372_bundle_invalid_with_v_0.json-      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml",
report_github372_bundle_invalid_with_v_0.json-      "messages": [
report_github372_bundle_invalid_with_v_0.json-        {
report_github372_bundle_invalid_with_v_0.json-          "severity": "WARNING",
report_github372_bundle_invalid_with_v_0.json-          "type": "info.integrity.unreferenced_file",
report_github372_bundle_invalid_with_v_0.json-          "message": "Skipping file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml due to not being selected as the bundle target"
--
report_github372_bundle_invalid_with_v_0.json:      "status": "SKIP",
report_github372_bundle_invalid_with_v_0.json-      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml",
report_github372_bundle_invalid_with_v_0.json-      "messages": [
report_github372_bundle_invalid_with_v_0.json-        {
report_github372_bundle_invalid_with_v_0.json-          "severity": "WARNING",
report_github372_bundle_invalid_with_v_0.json-          "type": "info.integrity.unreferenced_file",
report_github372_bundle_invalid_with_v_0.json-          "message": "Skipping file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml due to logical_identifier the same as selected collection but different version"
--
report_github372_bundle_invalid_with_v_1.json:      "status": "SKIP",
report_github372_bundle_invalid_with_v_1.json-      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml",
report_github372_bundle_invalid_with_v_1.json-      "messages": [
report_github372_bundle_invalid_with_v_1.json-        {
report_github372_bundle_invalid_with_v_1.json-          "severity": "WARNING",
report_github372_bundle_invalid_with_v_1.json-          "type": "info.integrity.unreferenced_file",
report_github372_bundle_invalid_with_v_1.json-          "message": "Skipping file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/bundle_kaguya_derived_2.xml due to not being selected as the bundle target"
--
report_github372_bundle_invalid_with_v_1.json:      "status": "SKIP",
report_github372_bundle_invalid_with_v_1.json-      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml",
report_github372_bundle_invalid_with_v_1.json-      "messages": [
report_github372_bundle_invalid_with_v_1.json-        {
report_github372_bundle_invalid_with_v_1.json-          "severity": "WARNING",
report_github372_bundle_invalid_with_v_1.json-          "type": "info.integrity.unreferenced_file",
report_github372_bundle_invalid_with_v_1.json-          "message": "Skipping file:/home/qchau/sandbox/validate/src/test/resources/github69/invalid/data_spectra/spectra_data_collection_inventory_2.xml due to logical_identifier the same as selected collection but different version"
```

qchaupds replied:
@jordanpadams The ExceptionType used now is INFO.

qchaupds replied:
Re-ran the above commands again and the severity is now INFO:

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 139 % grep -A 6 SKIP report_github372_bundle_invalid_with*.json | grep severity

report_github372_bundle_invalid_with_v_0.json-          "severity": "INFO",
report_github372_bundle_invalid_with_v_0.json-          "severity": "INFO",
report_github372_bundle_invalid_with_v_1.json-          "severity": "INFO",
report_github372_bundle_invalid_with_v_1.json-          "severity": "INFO",

The only troubling thing is the report file report_github372_bundle_invalid_without_v.json does not have the "SKIP" line.
Do you know why?

jordanpadams replied:
@qchaupds i think this problem is resovled now that we use the reporting mechanisms in the code?

qchaupds replied:
Yes, the printing of skipped files is done via the recordSkip() function now.  Please see latest comment.

jordanpadams reviewed (changes requested):
@qchaupds we are getting closer. see my comments inline. I have some other comments about the output, but I think that should go away once the changes are completed.

Commit:
Simplify logic of reporting a skip file in BundleManager.java and leave the logic on what to print to Report.java class

Commit:
Changed WARNING to INFO for ExceptionType

Commit:
Fix issues with bot complaining.

jordanpadams reviewed (changes requested):
@qchaupds getting closer. I made an update to `Report.java` to output skipped products all the time.

the only issue I am now seeing is we are not picking up the older version of bundle if I point validate at a directory for the bundle validation instead of a bundle.xml file.

jordanpadams committed:
Update to always output skipped

Commit:
Add record SKIP to report for bundle files that are ignored due to not latest version

Commit:
Merge branch '#372' of github.com:NASA-PDS/validate into #372

Commit:
Correct message

jordanpadams reviewed (changes requested):
@qchaupds closer, but still have an issue:

When validate runs against the directory, it picks up `bundle_kaguya_derived_2.XML` succesfully, but it seems to ignore the other skipped collection:
```
validate-2.1.0-SNAPSHOT/bin/validate --skip-content-validation -R pds4.bundle -t src/test/resources/github69/invalid/
...
  Product Validation Summary:
    9          product(s) passed
    0          product(s) failed
    1          product(s) skipped
...
```

vs. running against `bundle_kaguya_derived_2.XML` itself
```
validate-2.1.0-SNAPSHOT/bin/validate --skip-content-validation -R pds4.bundle -t src/test/resources/github69/invalid/bundle_kaguya_derived_2.XML
...
  Product Validation Summary:
    9          product(s) passed
    0          product(s) failed
    2          product(s) skipped
...
```

Commit:
Third responses to comments from pull request: Add reporting of collection files skipped in report when the target is a directory

1. Add new function containsExistingTarget() to inspect from a list of Target :src/main/java/gov/nasa/pds/tools/validate/BundleManager.java
2. Modify parameter in findOtherCollectionFiles() to receive a list of Target instead of a single URL in case there are more than 1 latest collections :src/main/java/gov/nasa/pds/tools/validate/BundleManager.java
3. Modify variable from otherBundleFilesList to otherCollectionFilesList to match purpose of function :src/main/java/gov/nasa/pds/tools/validate/BundleManager.java
4. Add reporting of collection files skipped in report when the target is a directory: src/main/java/gov/nasa/pds/tools/validate/BundleManager.java

Refs:

https://github.com/nasa-pds/validate/issues/372 Issues with logic for reading latest version of collections, and file read logging lost with v2.* of validate

Commit:
Check for null-ness before getting list size

jordanpadams committed:
Code cleanup

jordanpadams committed:
Merge branch 'main' into #372
